types/react/v17/index.d.ts

```diff
@@ -113,6 +113,9 @@ declare namespace React {
         key: string | null;
     }
 
+    /**
+     * @deprecated
```

Comment:
We don't need to deprecate it for the old types since we're not going to remove it from them.